### PR TITLE
Tests of new components of CoursePage - teacher edit screens

### PR DESCRIPTION
**Pull request by FlorianDinant**
# Pull Request Description  

This PR introduces a variety of important changes, enhancements, and tests to the project. The main focus is to improve the testing coverage and add key components that were previously missing or incomplete. Below is a detailed breakdown of the changes introduced in this PR:  

## **Tests for Existing Components**  
- Added comprehensive tests for the following components that were initially implemented in PR #71 but lacked proper test coverage:  
  - `AddAssignment`: Tests cover the functionality for creating assignments, form validations, and integration with Firebase updates.  
  - `AddMaterial`: Tests ensure proper handling of file uploads, metadata saving, and Firebase integration.  
  - `CourseParameters`: Verifies that this component handles course settings correctly, including toggling options and saving configurations.  
  - `SelectActionsCourse`: Ensures that action selection for courses behaves as expected, including triggering the correct events.  

## **Introduction of New Components and Corresponding Tests**  
- **`EditAssignment`**: A new component that enables professors to modify existing assignments within the course. Key features include:  
  - Real-time updates of assignment data within Firebase.  
  - User-friendly interface for editing assignment details such as due dates, descriptions, and file attachments.  
  - Tests include:  
    - Validation scenarios (e.g., missing required fields).  
    - Firebase data synchronization tests.  
    - UI functionality, such as field updates and feedback.  
- **`EditMaterial`**: Another essential addition for course management, this component allows professors to edit uploaded materials. Highlights include:  
  - Full synchronization with Firebase to ensure updates are persistent.  
  - Tests include:  
    - Edge cases (e.g., unsupported file types or incomplete submissions).  
    - Integration testing with the `courseActionsFunctions`.  
  - PLEASE note that for now we cannot upload documents such as PDFs etc... This will be don win the next sprint

## **Updates to Utility Functions**  
- **`courseActionsFunctions.js`**:  
  - Enhanced to support new operations related to editing and managing course data.  
  - Functions now include robust error handling and detailed logging to aid debugging.  
  - Modularized logic to improve readability and reusability, ensuring compatibility with new and existing components.  
  - Use of Toasts to inform the user the well or unwell behaviour of the action

## **Miscellaneous Enhancements**  
- Refactored some Firebase-related code to improve maintainability and efficiency, such as batching writes for better performance.  
- Updates to the UI of the course management components for a more cohesive user experience.  

### `CoursePage`: Teacher's View

Screenshot 2024-12-02 at 18 28 42
Screenshot 2024-12-02 at 18 31 43
Screenshot 2024-12-02 at 18 32 50

### `EditAssignment` Component: Displaying before and after editing an assignment.  

Screenshot 2024-12-02 at 18 31 57
Screenshot 2024-12-02 at 18 32 40

### `EditMaterial` Component: Highlighting the workflow for editing course materials.  

Screenshot 2024-12-02 at 18 33 16
Screenshot 2024-12-02 at 18 33 06

###  Test Coverage Report: Showing the increased test coverage achieved with this PR.  

Screenshot 2024-12-02 at 18 36 37

This PR not only improves the test coverage but also introduces critical components that enhance the course management experience for professors. Please review thoroughly and provide feedback to ensure all aspects of this PR meet the highest quality standards.  However, please note that I did not had the time to refractor perfectly the code so there is many parts that could be really clean. It will be done in a next sprint.

Thank you for your time!

### CODE REVIEW

File: edweiss-app/app/(app)/courses/[id]/index.tsx [hunk 1]
```diff
@@ -197,19 +208,69 @@ const CoursePage: ApplicationRoute = () => {
...
+		removeAssignmentAction(id as CourseID, assignmentID);
+		setAssignmentToEdit(null);
+		setModalEditAssignmentVisible(false);
+	}, [id]);
+
+	const removeMaterialCallback = useCallback((materialID: MaterialID) => {
+		removeMaterialAction(id as CourseID, materialID);
+		setMaterialToEdit(null);
+		setModalEditMaterialVisible(false);
+	}, [id]);
```
Comment (adamm-alaoui-pro) [inline edweiss-app/app/(app)/courses/[id]/index.tsx:267]: That's an interresting approach you have here.
I see that you provide kind of duplicate actions.
I don't know if it's possible to do it in typescript, but I would have a suggestion as an enhancement for the future, is to create an `AbstractCourseContentEditor` where you put all the common ground function and create 2 instances for the `Assignment` and `Material`, this way you reduce significantly boilerplate code and leverage the DRY principle.

File: edweiss-app/components/courses/AddAssignment.tsx
Comment (adamm-alaoui-pro) [inline edweiss-app/components/courses/AddAssignment.tsx:1]: So this must be the file you mentionned in the PR description that you modularized.
Great decision.

File: edweiss-app/__test__/components/courses/SelectActionsCourse.test.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        expect(screen.getByTestId(testIDs.addMaterialBouton)).toBeTruthy();
+        expect(screen.getByTestId(testIDs.addMaterialText)).toBeTruthy();
+    });
+
+    it('should call onOutsideClick when the screenTouchable is pressed', () => {
+        const screen = render(<SelectActionsCourse
+            onOutsideClick={mockOnOutsideClick}
+            onSelectAssignment={mockOnSelectAssignment}
+            onSelectMaterial={mockOnSelectMaterial}
+        />);
```
Comment (adamm-alaoui-pro) [inline edweiss-app/__test__/components/courses/SelectActionsCourse.test.tsx:78]: I see that you reuse a lot of this boilerplate code. 
I propose a create a renderScreen where you put the code and then call it inside.
 Here's my suggestion:

``` tsx
const renderScreen =  render(<SelectActionsCourse
            onOutsideClick={mockOnOutsideClick}
            onSelectAssignment={mockOnSelectAssignment}
            onSelectMaterial={mockOnSelectMaterial}
        />);
```

and then you only have to call it in the tests. This is a way of improvement, feel free to do it, if you feel like it make sens to you in your code base.

File: edweiss-app/__test__/components/courses/SelectActionsCourse.test.tsx
Comment (adamm-alaoui-pro) [inline edweiss-app/__test__/components/courses/SelectActionsCourse.test.tsx:1]: You did a great job in this test. The code is well structured. There are no general issues with the test in my opinion. I have pointed out a minor way of improvement. Have a glance to check-it out.

File: edweiss-app/__test__/utils/courses/coursesActionsFunctions.test.tsx
Comment (adamm-alaoui-pro) [inline edweiss-app/__test__/utils/courses/coursesActionsFunctions.test.tsx:1]: Hey there, great test, easy to read an understand.
I may have again a suggestion, to avoid duplicated code.
In order to check the Toast status, you could also generate a function that take as argument the `status` value and the expected strings, and generate the test. You can see it as an application as an application of the Builder DP. That's just a suggestion, the most important part is to have coverage and stress the correct functionality.

File: edweiss-app/components/courses/AssignmentComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+const AssignmentComponent: ReactComponent<EditAssignmentProps> = ({ mode, onSubmit, onDelete, assignment }) => {
+
+    const [name, setName] = React.useState<string>(assignment && mode == 'edit' ? assignment.data.name : "");
+    const [type, setType] = React.useState<AssignmentType>(assignment && mode == 'edit' ? assignment.data.type : 'quiz');
+    const [dueDate, setDueDate] = React.useState<Date>(assignment && mode == 'edit' ? Time.toDate(assignment.data.dueDate) : new Date());
+    const [nameChanged, setNameChanged] = useState<boolean>(false);
+    const [dateChanged, setDateChanged] = useState<boolean>(false);
+    const [timeChanged, setTimeChanged] = useState<boolean>(false);
+    const [showPickerDate, setShowPickerDate] = useState<boolean>(false);
+    const [showPickerTime, setShowPickerTime] = useState<boolean>(false);
```
Comment (adamm-alaoui-pro) [inline edweiss-app/components/courses/AssignmentComponent.tsx:92]: This is a great way to handle the codebase using use states. The things is that semantically the use states are not independant,. What I mean by this is context indenpant. They are all related to the `AssignementComponent` state.
In the futur, I would propose you this approach in order to pack things together and enhance code maintainability.

Refactor related states into a single `useReducer` or an object in `useState`. Here's what I mean:

```tsx
const [state, setState] = useState({
    name: assignment?.data.name || "",
    type: assignment?.data.type || "quiz",
    dueDate: assignment?.data.dueDate || new Date(),
    nameChanged: false,
    dateChanged: false,
    timeChanged: false,
    showPickerDate: false,
    showPickerTime: false,
});

const updateState = (key: string, value: any) => {
    setState((prev) => ({ ...prev, [key]: value }));
};

```

and you can use it this way to update the fields
```tsx

     updateState('name', 'new name');

```

File: edweiss-app/components/courses/AssignmentDisplay.tsx [hunk 4]
```diff
@@ -98,6 +111,7 @@ const AssignmentDisplay: ReactComponent<{ item: AssignmentWithColor, id: string,
                     swipeableRefs.current[index] = ref;
                 }}
                 renderRightActions={renderRightActions}
+                renderLeftActions={isTeacher ? renderLeftActions : undefined}
```
Comment (adamm-alaoui-pro) [inline edweiss-app/components/courses/AssignmentDisplay.tsx:114]: You made this only possible as being a professor. That's a great idea. That would have been fun if a student can `edit` it  😋.

File: edweiss-app/components/courses/CourseParameters.tsx
Comment (adamm-alaoui-pro) [inline edweiss-app/components/courses/CourseParameters.tsx:1]: That's okay with me.
I would just suggest in the future to enhance a bit the documentation.
This way we see with a glance to which button and textinput is the thing related. 
But that's alright with me, taste and colors.

**Review by adamm-alaoui-pro — APPROVED** [review]
To start of I would like to mention that you  have done a great job in this PR.

I appreciate the effort you put into modularizing stuff together. Especially the `AddAssignment` and `EditAssignement`. It happens that Sonar Cloud says some not idiot remarks (from time to time ...) Comming from Java one could have called it `AbstractEditorFactory ...`

You did a great job at testing   `AddAssignment`, `AddMaterial`, `CourseParameters` and `SelectActionsCourse`.

I may have a suggestion for the next sprint where you will upload the PDF, have a look at what @Gutomaia did in the `ShowTime` module to grasp if needed how the the `cloud-storage` get called.

I have tested your framework. I like the swipe to edit stuffs. It comes handy and natural to a user that discover this new functionality. Moreover, I see that you took some of the code provided in the `AbstractTodoEditor` regarding the `TimeDatePicker`, well done! I have a small question, do you think that it would be possible in the futur to add the `RichText` framework for the description. This way in case of an mathematical class one could illustrate with mathematical expressions. 

Regarding the coverage, that's really great. I just have a small remark. Could you explain me why are the part are not covered ? Are they constants, arguments pass, console.logs. It just for me to know if I could help you into getting those lines

The only pitfall that I would like to highlight is the size of the PR. That's really great what you did, but a PR with a >1KLoC is a bit hard to keep up with. Moreover, I agree with your end remark that some code could be cleaned and modularized a bit more. Looking forward into it for next sprint.

File: edweiss-app/__test__/components/courses/AssignmentComponent.test.tsx
[the remark this answers is not in the document]
Comment (FlorianDinant) [reply edweiss-app/__test__/components/courses/AssignmentComponent.test.tsx:1]: I already use `.toBeNull()` instead

File: edweiss-app/__test__/components/courses/MaterialComponent.test.tsx [hunk 5]
```diff
@@ -0,0 +1,361 @@
...
+            return children === text;
+        }
+        if (Array.isArray(children)) {
+            return children.some(child => hasTextInChildren(child, text));
+        }
+        if (children && typeof children === 'object' && children.props) {
+            return hasTextInChildren(children.props.children, text);
+        }
+        return false;
+    }
```
[the remark this answers is not in the document]
Comment (FlorianDinant) [reply edweiss-app/__test__/components/courses/MaterialComponent.test.tsx:207]: Yes exactly! It recursively checks that a text component is in it and displays `text`

File: edweiss-app/model/quizzes.ts [hunk 6]
```diff
@@ -38,7 +38,7 @@ namespace Quizzes {
         showResultToStudents: boolean;
         // inherits:
         // name: string
-        // type: AssignmentType = "quiz"|"submit"
+        // type: AssignmentType = "quiz"|"submission"
         // dueDate: TimeStamp
```
[the remark this answers is not in the document]
Comment (FlorianDinant) [reply edweiss-app/model/quizzes.ts:42]: Quizzes are not my part. I think Ferdinand works on it, I prefer let him refractor his stuff. I just decided to change the AssignmentType everywhere in the code to make it be "submission" instead of "submit" which is much better for understanding.

File: edweiss-app/utils/courses/coursesActionsFunctions.tsx [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+            text1: t(`course:failureAddMaterial`),
+        });
+        return { ...res, error: new Error(res.error) };
+    }
+    Toast.show({
+        type: 'success',
+        text1: t(`course:successAddMaterial`),
+    });
+    return res;
+}
```
[the remark this answers is not in the document]
Comment (FlorianDinant) [reply edweiss-app/utils/courses/coursesActionsFunctions.tsx:40]: Thank you for the advice, I take it into account

**Comment by adamm-alaoui-pro** [comment]
> > Regarding the coverage, that's really great. I just have a small remark. Could you explain me why are the part are not covered ? Are they constants, arguments pass, console.logs. It just for me to know if I could help you into getting those lines
> > The only pitfall that I would like to highlight is the size of the PR. That's really great what you did, but a PR with a >1KLoC is a bit hard to keep up with. Moreover, I agree with your end remark that some code could be cleaned and modularized a bit more. Looking forward into it for next sprint.
> 
> Can you be more precise? I have tester everything and got >80% coverage. Regarding the size of the PRs I will stop doing big PRs like that. I am tired on working 35 hours per week on it so the next PRs will have bettween 500 and 1000 lines modifs.

Hey, it's not about the coverage. I was just curious if you could explain me briefly the corner cases that were not captured. It's just a great to keep of which branch or statements were not covered to keep trace of it.

For instance, in my PRs I like to mention that this line is uncovered because it's due to the fact that it's a multi-line argument input, or that's due to a temporary `console.log` ...

The point of my request, is only to document the things so that, hope you understand.

File: edweiss-app/components/courses/AssignmentComponent.tsx (hunk 3 above)
Comment (FlorianDinant) [reply edweiss-app/components/courses/AssignmentComponent.tsx]: I slightly modified your version and integrated it, you can take a look if you want to. Thank you!

**Comment by FlorianDinant** [comment]
> > > Regarding the coverage, that's really great. I just have a small remark. Could you explain me why are the part are not covered ? Are they constants, arguments pass, console.logs. It just for me to know if I could help you into getting those lines
> > > The only pitfall that I would like to highlight is the size of the PR. That's really great what you did, but a PR with a >1KLoC is a bit hard to keep up with. Moreover, I agree with your end remark that some code could be cleaned and modularized a bit more. Looking forward into it for next sprint.
> > 
> > 
> > Can you be more precise? I have tester everything and got >80% coverage. Regarding the size of the PRs I will stop doing big PRs like that. I am tired on working 35 hours per week on it so the next PRs will have bettween 500 and 1000 lines modifs.
> 
> Before hand, I would like to thank you for the dedication to this project. We are all trying our best.
> 
> So, my main request was not about the coverage. I was just curious if you could explain me briefly the corner cases that were not captured. It's just a great to keep of which branch or statements were not covered to keep trace of it.
> 
> For instance, in my PRs I like to mention that this line is uncovered because it's due to the fact that it's a multi-line argument input, or that's due to a temporary `console.log` ...
> 
> The point of my request, is only to document the things so that, hope you understand.

The edges cases are principally swipe actions which are too difficult to be tested. But yes there is other cases that could be tested but since I achieved a good coverage I preferred to stop there since I have a lot of work in my other courses. Those cases could be tested if some coverage is missing by the end of the milestone tho

File: edweiss-app/app/(app)/courses/[id]/archive.tsx [hunk 8]
```diff
@@ -68,7 +68,7 @@ const ArchiveScreen: ApplicationRoute = () => {
 
                 {assignments.length > 0 ?
                     assignments.map((assignment) => (
-                        <AssignmentDisplay item={assignment.data} id={assignment.id} courseID={courseId} index={assignments.indexOf(assignment)} isSwipeable={false} key={assignment.data.name} />
+                        <AssignmentDisplay item={assignment.data} id={assignment.id} courseID={courseId} index={assignments.indexOf(assignment)} key={assignment.data.name} />
                     ))
```
Comment (Gwen678) [inline edweiss-app/app/(app)/courses/[id]/archive.tsx:72]: You can find index directly in map component 

`    assignments.map((assignment, index) => (`
it's faster 
for the key i think it's not a good idee to use the name because multiple assignment can have the same name. you can use the id

File: edweiss-app/components/courses/AssignmentComponent.tsx (hunk 3 above)
Comment (holoisme) [reply edweiss-app/components/courses/AssignmentComponent.tsx]: Hello,
This is bad practice as each `setState` would require to reconstruct the object.
This does not provide any advantage over @FlorianDinant 's method, as `updateState('name', 'new name')` has no advantage over `setName('new name')`
Also, this version does not provide any Intellisense or type safety.
`updateState("namr", 2)` is a valid call, this is obviously bad.
Please do not implement this.

**Comment by adamm-alaoui-pro** [comment]
> > > > Regarding the coverage, that's really great. I just have a small remark. Could you explain me why are the part are not covered ? Are they constants, arguments pass, console.logs. It just for me to know if I could help you into getting those lines
> > > > The only pitfall that I would like to highlight is the size of the PR. That's really great what you did, but a PR with a >1KLoC is a bit hard to keep up with. Moreover, I agree with your end remark that some code could be cleaned and modularized a bit more. Looking forward into it for next sprint.
> > > 
> > > 
> > > Can you be more precise? I have tester everything and got >80% coverage. Regarding the size of the PRs I will stop doing big PRs like that. I am tired on working 35 hours per week on it so the next PRs will have bettween 500 and 1000 lines modifs.
> > 
> > 
> > Before hand, I would like to thank you for the dedication to this project. We are all trying our best.
> > So, my main request was not about the coverage. I was just curious if you could explain me briefly the corner cases that were not captured. It's just a great to keep of which branch or statements were not covered to keep trace of it.
> > For instance, in my PRs I like to mention that this line is uncovered because it's due to the fact that it's a multi-line argument input, or that's due to a temporary `console.log` ...
> > The point of my request, is only to document the things so that, hope you understand.
> 
> The edges cases are principally swipe actions which are too difficult to be tested. But yes there is other cases that could be tested but since I achieved a good coverage I preferred to stop there since I have a lot of work in my other courses. Those cases could be tested if some coverage is missing by the end of the milestone tho

Perfect! That totally make sens. 
Thanks for sharing those informations.
I know what it's like to test swipe actions. (So imagine what it's like to have a 2 dimensional swipping 🤤 as in the todolist)

Your coverage is more than descent. I appreciate the amount of effort you involved into it.

File: edweiss-app/utils/courses/coursesActionsFunctions.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (holoisme) [reply edweiss-app/utils/courses/coursesActionsFunctions.tsx:40]: Again, this completely throws away any typesafety. Please use all Typescript features to ensure that invalid states and calls are unrepresentable.

**Review by Gwen678 — APPROVED** [review]
The code is well-structured and follows good practices, such as centralizing constants (icons, testIDs), which makes maintenance easier. The state management with useState is correct, but the nested conditions for rendering mode could be simplified to improve readability. Typing is generally well used, but the use of any for events should be avoided in favor of more explicit types. Some improvements could be made to enhance readability and accessibility, such as extracting certain conditions into variables or functions. It's good code, well done!

File: edweiss-app/__test__/components/courses/MaterialComponent.test.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (adamm-alaoui-pro) [reply edweiss-app/__test__/components/courses/MaterialComponent.test.tsx:207]: Amazing!

File: edweiss-app/locales/fr/course.json
[the remark this answers is not in the document]
Comment (adamm-alaoui-pro) [reply edweiss-app/locales/fr/course.json:1]: No worries, you are absolutly right. Let first finish and make sure that the app is stable then let add this feature.

File: edweiss-firebase/functions/package.json [hunk 9]
```diff
@@ -16,7 +16,7 @@
   "dependencies": {
     "dotenv": "^16.4.5",
     "firebase-admin": "^12.1.0",
-    "firebase-functions": "^5.0.0",
+    "firebase-functions": "^6.1.1",
```
[the remark this answers is not in the document]
Comment (adamm-alaoui-pro) [reply edweiss-firebase/functions/package.json:19]: TBH I have no real idea. Let keep it this way. We would see if it breaks something and will rollback to the previous version.

File: edweiss-app/__test__/utils/courses/coursesActionsFunctions.test.tsx
Comment (adamm-alaoui-pro) [reply edweiss-app/__test__/utils/courses/coursesActionsFunctions.test.tsx:1]: No  worries, that's more of a suggestion.

File: model/src/quizzes.ts [hunk 10]
```diff
@@ -38,7 +38,7 @@ namespace Quizzes {
         showResultToStudents: boolean;
         // inherits:
         // name: string
-        // type: AssignmentType = "quiz"|"submit"
+        // type: AssignmentType = "quiz"|"submission"
         // dueDate: TimeStamp
```
[the remark this answers is not in the document]
Comment (FlorianDinant) [reply model/src/quizzes.ts:42]: because "submission" is more explicit than "submit". So I decided to change all ocurrences in the project so I also modified the commented code

File: edweiss-firebase/functions/src/functions/course/updateMaterial.ts [hunk 11]
```diff
@@ -51,10 +51,13 @@ export const updateMaterial = onSanitizedCall(Functions.updateMaterial, {
...
     }
 
+    assertIsBetween(materialData.title.length, 1, MAX_MATERIAL_TITLE_LENGTH, "material_title_too_long");
+    assertIsBetween(materialData.description.length, 0, MAX_MATERIAL_DESCRIPTION_LENGTH, "material_description_too_long");
+
     // Prepare updated fields
     const updatedFields: Partial<Material> = {};
     if (materialData.title) updatedFields.title = materialData.title;
-    if (materialData.description) updatedFields.description = materialData.description;
+    if (materialData.description || materialData.description === '') updatedFields.description = materialData.description;
```
[the remark this answers is not in the document]
Comment (FlorianDinant) [reply edweiss-firebase/functions/src/functions/course/updateMaterial.ts:60]: I think I prefer use the current because `materialData.description !== undefined` does not test if it is null. However `materialData.description` checks that it is neither null nor undefined, nor empty nor 0.

File: model/src/school/courses.ts [hunk 12]
```diff
@@ -87,4 +95,11 @@ export namespace Course_functions {
...
-}
\ No newline at end of file
+}
+
+export type UpdateCourseArgs = {
+	name: string,
+	description: string,
+	credits: Credits,
+	section: Section,
+}
```
[the remark this answers is not in the document]
Comment (FlorianDinant) [reply model/src/school/courses.ts:105]: It is already done in the firebase functions so no need to add checks here since onSanitizedCall and all the other checks inside already handle those cases. Also, the `CourseParameters.tsx` file does not allow the user to click on the Update Changes button if those properties are not satisfied so no need to do that.

File: edweiss-firebase/functions/src/functions/course/updateAssignment.ts [hunk 13]
```diff
@@ -62,7 +61,7 @@ export const updateAssignment = onSanitizedCall(Functions.updateAssignment, {
     const updatedFields: Partial<Assignment> = {};
     if (assignmentData.name) {
         assertNonEmptyString(assignmentData.name, "invalid_name");
-        assertIsBetween(assignmentData.name.length, 1, MAX_NAME_LENGTH, "invalid_name");
+        assertIsBetween(assignmentData.name.length, 1, MAX_ASSIGNMENT_NAME_LENGTH, "invalid_name");
         updatedFields.name = assignmentData.name as string;
     }
```
[the remark this answers is not in the document]
Comment (FlorianDinant) [reply edweiss-firebase/functions/src/functions/course/updateAssignment.ts:66]: Yes you are right !

File: edweiss-app/components/courses/CourseParameters.tsx [hunk 14]
```diff
@@ -16,31 +17,53 @@ import FancyTextInput from '../input/FancyTextInput';
...
+    closeIcon: 'close-icon',
+    title: 'title',
+    message: 'message',
+    scrollView: 'scroll-view',
+    nameDescriptionSectionView: 'name-description-section-view',
+    nameInput: 'name-input',
+    descriptionInput: 'description-input',
+    sectionInput: 'section-input',
+    creditsComponentView: 'credits-component-view',
+    descreaseCreidtsButton: 'descrease-credits-button',
```
[the remark this answers is not in the document]
Comment (FlorianDinant) [reply edweiss-app/components/courses/CourseParameters.tsx:38]: What do you mean? A file where every testIds of every files are displayed ?!!?
 Imo I think it not a good idea to do that since those testIds are specific to this file and move them elsewhere is just annoying if we want to modify on testId.

File: edweiss-app/components/courses/AddAssignment.tsx
[the remark this answers is not in the document]
Comment (FlorianDinant) [reply edweiss-app/components/courses/AddAssignment.tsx:1]: As said in the PR description, I merged `AddAssignment` and `EditAssignment` into one single filed called AssignmentComponent. Idem for `AddMaterial`

File: edweiss-app/app/(app)/courses/[id]/archive.tsx (hunk 8 above)
Comment (FlorianDinant) [reply edweiss-app/app/(app)/courses/[id]/archive.tsx:72]: Yes you are right I don't know why this has been on the main branch all this time
